Backend/CourseService.cs

```diff
@@ -50,9 +52,38 @@ public async Task<Course> FindByNameAsync(string name)
             return await _courseCollection.Find(x => x.CourseName.Equals(name.ToLower())).FirstOrDefaultAsync();
         }
 
-        public List<Course> FindByStartName(string courseStartsWith)
+        private string RemoveDiacritics(string text)
         {
-            return _courseCollection.Find(x => x.CourseName.ToLower().Contains(courseStartsWith.ToLower())).ToList();
+            var normalizedString = text.Normalize(NormalizationForm.FormD);
+            var stringBuilder = new StringBuilder();
+
+            foreach (var c in normalizedString)
+            {
+                var unicodeCategory = CharUnicodeInfo.GetUnicodeCategory(c);
+                if (unicodeCategory != UnicodeCategory.NonSpacingMark)
+                {
+                    stringBuilder.Append(c);
+                }
+            }
+
+            return stringBuilder.ToString().Normalize(NormalizationForm.FormC);
+        }
+
+        public List<Course> FindByStartName(string courseStartsWith, string personalNumber)
```

Comment:
Maybe add something like if courseStartsWith length is more than 3 than do this just suggestion, because 1,2 or 3 characters dont say much

Reply:
Good point, but I think that already 3 characters is enough to find suggested subjects.